### PR TITLE
Allow all service account to bump version

### PR DESCRIPTION
**Pull request by chzhanpeng**
**Affects:** \<api-audit-version-number>.

### CODE REVIEW

File: src/main/java/com/capitalone/dashboard/common/CommonCodeReview.java [hunk 1]
```diff
@@ -181,19 +177,18 @@ public static boolean checkForServiceAccount(String userLdapDN, ApiSettings sett
...
-        for (String serviceAccount:allowedServiceAccounts.keySet()) {
-            String fileNames = allowedServiceAccounts.get(serviceAccount);
-            for (String s : fileNames.split(",")) {
-                if (serviceAccount.equalsIgnoreCase(author) && findFileMatch(s, commitFiles)){
-                    isValidServiceAccount = true;
-                }
+    public static boolean isFileTypeWhitelisted(Commit commit, ApiSettings settings) {
+        List<String> whitelistedFiles = settings.getDirectCommitWhitelistedFiles();
+        Stream<String> combinedStream
+                = Stream.of(commit.getFilesAdded(), commit.getFilesModified(),commit.getFilesRemoved()).filter(Objects::nonNull).flatMap(Collection::stream);
```
Comment (nireeshT) [inline src/main/java/com/capitalone/dashboard/common/CommonCodeReview.java:183]: @chzhanpeng  commit.getFilesAdded(), commit.getFilesModified(),commit.getFilesRemoved() could be null. Is this code null safe?

File: src/test/java/com/capitalone/dashboard/common/CommonCodeReviewTest.java [hunk 2]
```diff
@@ -140,6 +119,19 @@ private Commit makeCommit() {
...
+    private Commit makeCommitWhitelistedFiles() {
+        Commit c = new Commit();
+        c.setId(ObjectId.get());
+        c.setScmCommitLog("Merge branch master into branch");
+        c.setScmAuthor("hygieiaUser");
+        c.setScmAuthorLDAPDN("CN=hygieiaUser,OU=Service Accounts,DC=basic,DC=ds,DC=industry,DC=com");
+        c.setScmCommitTimestamp(100000000);
+        c.setScmAuthorLogin("hygieiaUser");
+        c.setFilesAdded(Arrays.asList("readme.md", "pom.xml"));
         return c;
```
Comment (nireeshT) [inline src/test/java/com/capitalone/dashboard/common/CommonCodeReviewTest.java:135]: @chzhanpeng  Add test case for commit.getFilesAdded(), commit.getFilesModified(),commit.getFilesRemoved() null

File: src/main/java/com/capitalone/dashboard/common/CommonCodeReview.java (hunk 1 above)
Comment (chzhanpeng) [reply src/main/java/com/capitalone/dashboard/common/CommonCodeReview.java:183]: Yes, `for (String file : updatedFiles) {` is null safe

Comment (chzhanpeng) [reply src/main/java/com/capitalone/dashboard/common/CommonCodeReview.java:183]: Yes

File: src/test/java/com/capitalone/dashboard/common/CommonCodeReviewTest.java (hunk 2 above)
Comment (chzhanpeng) [reply src/test/java/com/capitalone/dashboard/common/CommonCodeReviewTest.java:156]: Added

**Pull request merged**